src/app/clusters/bindings/PendingNotificationMap.cpp

```diff
@@ -79,11 +79,15 @@ CHIP_ERROR PendingNotificationMap::FindLRUConnectPeer(FabricIndex * fabric, Node
     return CHIP_ERROR_NOT_FOUND;
 }
 
-void PendingNotificationMap::AddPendingNotification(uint8_t bindingEntryId, void * context)
+void PendingNotificationMap::AddPendingNotification(uint8_t bindingEntryId, PendingNotificationContext * context)
 {
     RemoveEntry(bindingEntryId);
     mPendingBindingEntries[mNumEntries] = bindingEntryId;
     mPendingContexts[mNumEntries]       = context;
+    if (context)
```

Comment:
When would `context` be null here?

Reply:
In terms of the current `BindingManager` implementation, most likely never, however there is nothing that would prevent someone from doing it. For example the tests are passing nullptr to the `AddPendingNotification`: https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/TestPendingNotificationMap.cpp#L68. I could align that, but I think it has some sense to not allocate `PendingNotificationContext` object if it will point to the nullptr context, as actually there is nothing to pass and nothing to release. I thought that some additional checks will not harm, but protect from future bugs.

Reply:
> but I think it has some sense to not allocate PendingNotificationContext object if it will point to the nullptr context

But we're allocating in that case right now, right?

Anyway, if we have tests passing null here, OK.